src/test/java/org/apache/datasketches/kll/KllItemsSketchSerDeTest.java

```diff
@@ -0,0 +1,105 @@
+    assertEquals(bytes.length, sketch1.getSerializedSizeBytes());
+    assertTrue(sketch2.isEmpty());
+    assertEquals(sketch2.getNumRetained(), sketch1.getNumRetained());
+    assertEquals(sketch2.getN(), sketch1.getN());
+    assertEquals(sketch2.getNormalizedRankError(false), sketch1.getNormalizedRankError(false));
+    try { sketch2.getMinItem(); fail(); } catch (SketchesArgumentException e) {}
+    try { sketch2.getMaxItem(); fail(); } catch (SketchesArgumentException e) {}
+    assertEquals(sketch2.getSerializedSizeBytes(), sketch1.getSerializedSizeBytes());
+  }
+
+  @Test
+  public void serializeDeserializeOneValue() {
+    final KllItemsSketch<String> sketch1 = KllItemsSketch.newHeapInstance(20, Comparator.naturalOrder(), serDe);
+    sketch1.update(" 1");
+    final byte[] bytes = sketch1.toByteArray();
+    final KllItemsSketch<String> sketch2 = KllItemsSketch.heapify(Memory.wrap(bytes), Comparator.naturalOrder(), serDe);
+    assertEquals(bytes.length, sketch1.getSerializedSizeBytes());
+    assertFalse(sketch2.isEmpty());
+    assertEquals(sketch2.getNumRetained(), 1);
+    assertEquals(sketch2.getN(), 1);
+    assertEquals(sketch2.getNormalizedRankError(false), sketch1.getNormalizedRankError(false));
+    assertEquals(sketch2.getMinItem(), " 1");
+    assertEquals(sketch2.getMaxItem(), " 1");
+    assertEquals(sketch2.getSerializedSizeBytes(), sketch2.getSerializedSizeBytes());
+  }
+
+  @Test
+  public void serializeDeserialize() {
+    final KllItemsSketch<String> sketch1 = KllItemsSketch.newHeapInstance(Comparator.naturalOrder(), serDe);
+    final int n = 1000;
+    for (int i = 0; i < n; i++) {
+      sketch1.update(intToFixedLengthString(i, 4));
+    }
+    final byte[] bytes = sketch1.toByteArray();
+    final KllItemsSketch<String> sketch2 = KllItemsSketch.heapify(Memory.wrap(bytes), Comparator.naturalOrder(), serDe);
+    assertEquals(bytes.length, sketch1.getSerializedSizeBytes());
+    assertFalse(sketch2.isEmpty());
+    assertEquals(sketch2.getNumRetained(), sketch1.getNumRetained());
+    assertEquals(sketch2.getN(), sketch1.getN());
+    assertEquals(sketch2.getNormalizedRankError(false), sketch1.getNormalizedRankError(false));
+    assertEquals(sketch2.getMinItem(), sketch1.getMinItem());
+    assertEquals(sketch2.getMaxItem(), sketch1.getMaxItem());
+    assertEquals(sketch2.getSerializedSizeBytes(), sketch1.getSerializedSizeBytes());
+  }
+
+  //no cross language tests yet
+
+  @Test(groups = {"generate"})
+  public void generateBinariesForCompatibilityTesting() throws Exception {
+    final int[] nArr = {0, 1, 10, 100, 1_000, 10_000, 100_000, 1_000_000};
+    for (int n: nArr) {
+      final int digits = numDigits(n);
+      final KllItemsSketch<String> sketch = KllItemsSketch.newHeapInstance(Comparator.naturalOrder(), serDe);
+      for (int i = 0; i < n; i++) sketch.update(intToFixedLengthString(i, digits));
+      try (final FileOutputStream file = new FileOutputStream("kll_items_n" + n + ".sk")) {
```

Comment:
I would suggest naming the files "kll_string_n" to be consistent with the naming in other places

Reply:
Done